omod/src/main/java/org/openmrs/module/coreapps/page/controller/MergeVisitsPageController.java

```diff
@@ -52,27 +64,58 @@ public Object controller(@RequestParam("patientId") Patient patient,
 
     public String post(@RequestParam("patientId") Patient patient,
                        @RequestParam("mergeVisits") List<Integer> mergeVisits,
-					   @RequestParam(value = "returnUrl", required = false) String returnUrl,
+                       @RequestParam(value = "returnUrl", required = false) String returnUrl,
                        @SpringBean AdtService service,
                        UiUtils ui,
-                       HttpServletRequest request, PageModel model){
-
+                       HttpServletRequest request, PageModel model) {
+        List<String> mergedVisitsIDs = new ArrayList<String>();
         if (patient.isVoided() || patient.isPersonVoided()) {
             return "redirect:" + ui.pageLink("coreapps", "patientdashboard/deletedPatient",
-					SimpleObject.create("patientId", patient.getId().toString() ));
+                    SimpleObject.create("patientId", patient.getId().toString()));
         }
 
-        if (mergeVisits!=null && mergeVisits.size() > 0 ){
+        if (mergeVisits != null && mergeVisits.size() > 0) {
             Visit mergedVisit = service.mergeConsecutiveVisits(mergeVisits, patient);
-            if (mergedVisit != null){
+            if (mergedVisit != null) {
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_INFO_MESSAGE, ui.message("coreapps.task.mergeVisits.success"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
-            }else{
+            } else {
                 request.getSession().setAttribute("emr.errorMessage", ui.message("coreapps.task.mergeVisits.error"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
             }
+
+            for (int i = 1; i < mergeVisits.size(); i++) {
+                mergedVisitsIDs.add(mergeVisits.get(i).toString());
+            }
         }
 
-		return "redirect:" + ui.pageLink("coreapps", "mergeVisits", SimpleObject.create("patientId", patient.getId(), "returnUrl", returnUrl));
+
+        return "redirect:" + ui.pageLink("coreapps", "mergeVisits", SimpleObject.create("patientId", patient.getId(), "returnUrl", returnUrl, "mergedVisitsIDs", mergedVisitsIDs));
```

Comment:
`mergedVisitsIDs` → `mergedVisitIds`(in general btw, across your changes)

* VIsit singular
* IDs plural but written `Ids` in code.